ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/assertions/CommandDeduplicationAssertions.scala

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.testtool.infrastructure.assertions
+
+import java.time.Instant
+
+import com.daml.api.util.DurationConversion
+import com.daml.ledger.api.testtool.infrastructure.Assertions.{assertDefined, fail}
+import com.daml.ledger.api.testtool.infrastructure.participant.CompletionResponse
+import com.daml.ledger.api.v1.completion.Completion
+import com.daml.ledger.api.v1.experimental_features.CommandDeduplicationPeriodSupport.{
+  DurationSupport,
+  OffsetSupport,
+}
+import com.daml.lf.data.Ref
+import com.google.protobuf.duration.Duration
+
+object CommandDeduplicationAssertions {
+  def assertDeduplicationDuration(
+      requestedDeduplicationDuration: Duration,
+      previousSubmissionSendTime: Instant,
+      completionReceiveTime: Instant,
+      completion: Completion,
+      durationSupport: DurationSupport,
+  ): Unit = {
+    val requestedDuration = DurationConversion.fromProto(requestedDeduplicationDuration)
+    durationSupport match {
+      case DurationSupport.DURATION_NATIVE_SUPPORT =>
+        val reportedDurationProto = assertDefined(
+          completion.deduplicationPeriod.deduplicationDuration,
+          "No deduplication duration has been reported",
+        )
+        val reportedDuration = DurationConversion.fromProto(reportedDurationProto)
+        assert(
+          reportedDuration.compareTo(requestedDuration) >= 0,
```

Comment:
If you import `scala.math.Ordering.Implicits._`, you can use `>=` here.

also applies to other calls to `compareTo`.